packages/foundry/contracts/test/modules/RepayCloseModule.t.sol

```diff
@@ -0,0 +1,124 @@
+}
+
+contract RepayCloseModuleTest is ZeroTest {
+    function testOnlyBorrowAndPoolVault() public {
+        console.log("can only be used with Borrow and Pool vaults");
+        // Provide USDC ilkId instead of DAI
+        (vaultId, ) = ladle.build(seriesId, otherIlkId, 0);
+        deal(address(usdc), address(this), WAD * 10000);
+        DataTypes.Vault memory vault = cauldron.vaults(vaultId);
+        usdc.approve(address(ladle), WAD * 10000);
+        usdc.transfer(otherJoin, WAD * 10000);
+        ladle.pour(vaultId, vault.owner, 1e18 * 10000, 1e18 * 5000);
+        
+        vm.prank(address(ladle));
+        vm.expectRevert("Only for Borrow and Pool");
+        ILadleCustom(address(ladle)).moduleCall(
+            address(module),
+            abi.encodeWithSelector(module.repayFromLadle.selector, vaultId, foo, bar)
+        );
+    }
+}
+
+contract WithVaultProvisioned is ZeroTest {
+    function setUp() public override {
+        super.setUp();
+        // create vault
+        (vaultId, ) = ladle.build(seriesId, ilkId, 0);
+        // provide tokens
+        deal(address(dai), address(this), WAD * 2);
+        deal(address(dai), address(ladle), WAD);
+        deal(address(fyDAI), address(ladle), WAD);
+        // provision vault
+        DataTypes.Vault memory vault = cauldron.vaults(vaultId);
+        dai.approve(address(ladle), WAD * 2);
+        dai.transfer(join, WAD * 2);
+        ladle.pour(vaultId, vault.owner, 1e18 * 2, 1e18);
+    }
+
+    function testRepayFromLadle() public {
+        console.log("Can repay from ladle");
+        uint256 joinBalanceBefore = dai.balanceOf(address(join));
+        uint256 baseBalanceBefore = dai.balanceOf(address(foo));
+                
+        vm.prank(address(ladle));
+        bytes memory data = ILadleCustom(address(ladle)).moduleCall(
+            address(module),
+            abi.encodeWithSelector(module.repayFromLadle.selector, vaultId, foo, bar)
+        );
+        uint256 repaid = abi.decode(data, (uint256));
+
+        assertEq(repaid, WAD);
+        assertEq(joinBalanceBefore, dai.balanceOf(address(join)) + WAD);
+        assertEq(baseBalanceBefore, dai.balanceOf(address(foo)) - WAD);
+        assertEq(cauldron.balances(vaultId).ink, WAD);
+        assertEq(cauldron.balances(vaultId).art, 0);
+    }
+
+    function testCloseFromLadle() public {
+        console.log("Can close from ladle");
+        uint256 joinBalanceBefore = dai.balanceOf(address(join));
+        uint256 baseBalanceBefore = dai.balanceOf(address(foo));
+
+        vm.prank(address(ladle));
+        bytes memory data = ILadleCustom(address(ladle)).moduleCall(
+            address(module),
+            abi.encodeWithSelector(module.closeFromLadle.selector, vaultId, foo, bar)
+        );
+        uint256 repaid = abi.decode(data, (uint256));
+
+        assertEq(repaid, WAD);
+        assertEq(joinBalanceBefore, dai.balanceOf(address(join)));
+        assertEq(baseBalanceBefore, dai.balanceOf(address(foo)) - WAD);
```

Comment:
You are still not testing that the unused dai go to `bar`. To do so, you just need to deal more Dai to the Ladle than debt is there in the vault we are repaying.